src/Files.App/Helpers/AdaptiveLayoutHelpers.cs

```diff
@@ -3,124 +3,128 @@
-				{
-					layoutGridView();
-				}
-				else
-				{
-					layoutDetails();
-				}
-
-				return true;
 			}
+		}
+
+		private static Layouts GetAdaptiveLayout(string path, IList<ListedItem> filesAndFolders)
+		{
+			var pathLayout = GetPathLayout(path);
+			if (pathLayout is not Layouts.None)
+				return pathLayout;
+
+			return GetContentLayout(filesAndFolders);
+		}
+
+		private static Layouts GetPathLayout(string path)
+		{
+			var iniPath = IO.Path.Combine(path, "desktop.ini");
+
+			var iniContents = NativeFileOperationsHelper.ReadStringFromFile(iniPath)?.Trim();
+			if (string.IsNullOrEmpty(iniContents))
+				return Layouts.None;
+
+			var parser = new IniParser.Parser.IniDataParser();
+			parser.Configuration.ThrowExceptionsOnError = false;
+			var data = parser.Parse(iniContents);
+			if (data is null)
+				return Layouts.None;
+
+			var viewModeSection = data.Sections.FirstOrDefault(IsViewState);
+			if (viewModeSection is null)
+				return Layouts.None;
 
-			return false;
+			var folderTypeKey = viewModeSection.Keys.FirstOrDefault(IsFolderType);
+			if (folderTypeKey is null)
+				return Layouts.None;
+
+			return folderTypeKey.Value switch
+			{
+				"Pictures" => Layouts.Grid,
+				"Videos" => Layouts.Grid,
+				_ => Layouts.Detail,
+			};
+
+			static bool IsViewState(SectionData data)
+				=> "ViewState".Equals(data.SectionName, StringComparison.OrdinalIgnoreCase);
+
+			static bool IsFolderType(KeyData data)
+				=> "FolderType".Equals(data.KeyName, StringComparison.OrdinalIgnoreCase);
+		}
+
+		private static Layouts GetContentLayout(IList<ListedItem> filesAndFolders)
+		{
+			int itemCount = filesAndFolders.Count;
+			if (filesAndFolders.Count is 0)
+				return Layouts.None;
+
+			float folderPercentage = 100f * filesAndFolders.Count(IsFolder) / itemCount;
+			float imagePercentage = 100f * filesAndFolders.Count(IsImage) / itemCount;
+			float mediaPercentage = 100f * filesAndFolders.Count(IsMedia) / itemCount;
+			float miscPercentage = 100f - (folderPercentage + imagePercentage + mediaPercentage);
+
+			if (folderPercentage + miscPercentage > LargeThreshold)
+				return Layouts.Detail;
+			if (imagePercentage > ExtraLargeThreshold)
+				return Layouts.Grid;
+			if (imagePercentage <= MediumThreshold)
+				return Layouts.Detail;
+			if (100f - imagePercentage <= SmallThreshold)
+				return Layouts.Detail;
+			if (folderPercentage + miscPercentage <= ExtraSmallThreshold)
+				return Layouts.Detail;
+			return Layouts.Grid;
+
+			static bool IsFolder(ListedItem item)
+				=> item.PrimaryItemAttribute is StorageItemTypes.Folder;
+
+			static bool IsImage(ListedItem item)
+				=> !string.IsNullOrEmpty(item.FileExtension)
+				&& ImagePreviewViewModel.ContainsExtension(item.FileExtension.ToLowerInvariant());
+
+			static bool IsMedia(ListedItem item)
+				=> !string.IsNullOrEmpty(item.FileExtension)
+				&& MediaPreviewViewModel.ContainsExtension(item.FileExtension.ToLowerInvariant());
```

Comment:
This is out of this PR, but we should get all this `IsFolder`, `IsImage` and the likes inside a common helper and mutualize its use.

Reply:
Actually such an helper exists : `FileExtensionHelpers`.
Would be a good opportunity to enrich it!

Reply:
There is a difference in the behavior of these 2 methods. FileExtensionHelpers checks 4 extensions to display some commands (Rotate, ...). ImagePreviewViewModel checks 8 extensions to display the item as an image in the preview. It makes sense that .ico is displayed as an image in Preview but cannot be set as Wallpaper. For our case, it will be necessary to redefine the criteria (a file made up of videos is displayed in detail instead of grid). This pr was made to facilitate these behavior changes with a more readable and modifiable code. Maybe add a HasThumbnail property.

Reply:
Alright, this is something we want to keep track of, I'll copy part of it to #10831 since it may feed the discussion.